docs/source/trigger.rst

```diff
@@ -1,5 +1,17 @@
 Triggers
 ========
 
+A ``Trigger`` is the main type of user input plugins will see.
+
 .. autoclass:: sopel.trigger.Trigger
     :members:
+
+
+PreTriggers
+===========
```

Comment:
Even though Sphinx is usually smart with hierarchy, I'd recommend against using the same level as the main title.

Here, I'd see:

```
========
Triggers
========

.. autoclass:: sopel.trigger.Trigger

PreTriggers
===========

.. autoclass:: sopel.trigger.PreTrigger
```

Reply:
Does this hierarchy look fine to you? It seems OK to me, unless we're to consider a `PreTrigger` as subordinate to a `Trigger`. Which it is, kind of…but only functionally—the types are siblings in the code.

![image](https://user-images.githubusercontent.com/164140/70758164-aba64a80-1d07-11ea-8170-32ba85880cfe.png)

Put another way, I think we either keep both headings at the same level (optionally with a new top-level heading unifying the two, if your worry is actually about Sphinx getting confused somehow), or eliminate the "PreTriggers" heading and have them both under "Triggers".

Reply:
For completeness: The proposed heading structure makes "PreTriggers" a second-level heading, subordinate to the "Triggers" at page top.

![image](https://user-images.githubusercontent.com/164140/70758969-47d15100-1d0a-11ea-81fa-0036cab7ae0a.png)

In both cases, the page title remains "Triggers".

Reply:
Found a tiny problem with having two top-level headings on the page while working on `trigger`'s doc overhaul (still WIP): Sphinx generates navigation links for both "Trigger" and "PreTrigger". Didn't notice that before.

But my opinion stands: Either we add a new top-level heading above the two, or put both classes under the single top-level "Triggers" heading. (I'm open to other suggestions, of course. I've been 200% distracted during all Sopel stuff I did today, including this!)

Reply:
I'd say you can leave it as it was: Trigger is the page's title (until we find something better?), and both classes are under this title, without any subdivision. So:

```
Trigger
=======

.. class:: Trigger

.. class:: PreTrigger
```